### PR TITLE
send mail recommend product

### PR DESCRIPTION
**Pull request by ghost**
## Description
Send mail recommend product
## Reference Requirements
- Task ID: TI-1

## Reference Isuees
#{issue_id}

## Migrations
YES | NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: app/Mail/SendEmailRecommends.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+     *
+     * @return $this
+     */
+    public function build()
+    {
+        return $this->view('email.send-mail-recommend')
+        ->with([
+            'user' => $this->user,
+            'products' => $this->productRecommend,
+        ]);
```
Comment (at-thaitran) [inline app/Mail/SendEmailRecommends.php]: tab vo

File: resources/views/email/send-mail-recommend.blade.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<body>
+    <div>      
+        <h1>Welcome To The Website Food-Drink</h1>
+        <h3> {{$user->user->email}} </h3>
+        @foreach($products as $product)
+            <div>
+                <ul>        
+                    <li> {{ __('product.admin.create.name') }}: {{$product->name}}</li>
+                    <li> {{ __('product.admin.create.price') }}: {{$product->price}}</li>
+                    <li> {{ __('product.admin.create.preview') }}: {{$product->preview}}</li>
```
Comment (at-thaitran) [inline resources/views/email/send-mail-recommend.blade.php]: loi convension `{{ $... }}`

**Comment by at-thaitran** [comment]
LGTM

File: app/Console/Commands/SendEmailRecommend.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+use App\Jobs\SendEmailRecommendJob;
+
+class SendEmailRecommend extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'recommend-product:send-mail-user';
```
Comment (at-quangtran) [inline app/Console/Commands/SendEmailRecommend.php]: recommend-product:user

File: app/Jobs/SendEmailRecommendJob.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $users = Order::with('user')->get()->unique('user_id');
+        foreach ($users as $user) {
+            $orderProduct = OrderDetail::with('product')
```
Comment (at-quangtran) [inline app/Jobs/SendEmailRecommendJob.php]: có cần with('product')? trong bảng có trường product_id ko?

File: app/Jobs/SendEmailRecommendJob.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                ->whereIn('order_id', Order::where('user_id', $user->user_id)->get()->pluck('id'))
+                ->get()->unique('product_id')->pluck('product_id');
+            $category = Product::whereIn('id', $orderProduct)->get()->unique('category_id')->pluck('category_id');
+            $productRecommend = Product::with('category', 'images')->whereNotIn('id', $orderProduct)->whereIn('category_id', $category)->get();
```
Comment (at-quangtran) [inline app/Jobs/SendEmailRecommendJob.php]: có cần with('category', 'images') ko ? trong mail ko thấy sử dụng đến ?

**Review by mvn-quangtran-dn — APPROVED** [review]
LGTM

**Pull request merged**